norman_utils_external/file_utils.py

```diff
@@ -6,19 +6,90 @@
 
 
 class FileUtils(metaclass=Singleton):
+    """
+    Utility class for performing file-related operations, including
+    buffer-size detection, file type inference from binary headers,
+    and UTF-8 detection.
+
+    This class is implemented as a singleton to ensure consistent and
+    efficient use across the system, especially for repetitive header
+    inspections and file classification logic.
+
+    **File Type Detection**
+    The `get_file_type` method reads a portion of the file and determines
```

Comment:
Contents: should only contain class related data. Move method specific data to the method docstring.